File: client/java/src/main/java/com/be/mesh/client/system/MeshEvaluator.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.system;

import com.be.mesh.client.annotate.SPI;
import com.be.mesh.client.mpc.ServiceProxy;
import com.be.mesh.client.prsim.Evaluator;
import com.be.mesh.client.struct.Page;
import com.be.mesh.client.struct.Paging;
import com.be.mesh.client.struct.Script;

import java.util.List;
import java.util.Map;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public class MeshEvaluator implements Evaluator {

    private final Evaluator evaluator = ServiceProxy.proxy(Evaluator.class);

    @Override
    public String compile(Script script) {
        return evaluator.compile(script);
    }

    @Override
    public String exec(String code, Map<String, String> args, String dft) {
        return evaluator.exec(code, args, dft);
    }

    @Override
    public List<Script> dump(Map<String, String> feature) {
        return evaluator.dump(feature);
    }

    @Override
    public Page<Script> index(Paging index) {
        return evaluator.index(index);
    }
}

File: client/golang/system/mesh_evaluator.go
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

package system

import (
	"context"
	"github.com/be-io/mesh/client/golang/macro"
	"github.com/be-io/mesh/client/golang/prsim"
	"github.com/be-io/mesh/client/golang/types"
)

func init() {
	var _ prsim.Evaluator = new(MeshEvaluator)
	macro.Provide(prsim.IEvaluator, new(MeshEvaluator))
}

type MeshEvaluator struct {
}

func (that *MeshEvaluator) Att() *macro.Att {
	return &macro.Att{Name: macro.MeshSys}
}

func (that *MeshEvaluator) Compile(ctx context.Context, script *types.Script) (string, error) {
	return aware.Evaluator.Compile(ctx, script)
}

func (that *MeshEvaluator) Exec(ctx context.Context, code string, args map[string]string, dft string) (string, error) {
	return aware.Evaluator.Exec(ctx, code, args, dft)
}

func (that *MeshEvaluator) Dump(ctx context.Context, feature map[string]string) ([]*types.Script, error) {
	return aware.Evaluator.Dump(ctx, feature)
}

func (that *MeshEvaluator) Index(ctx context.Context, index *types.Paging) (*types.Page[*types.Script], error) {
	return aware.Evaluator.Index(ctx, index)
}
